Build fails with GHC 8.8

```
Building library for units-2.4.1.1..
[ 1 of 19] Compiling Data.Metrology.Set ( Data/Metrology/Set.hs, dist/build/Data/Metrology/Set.o )
[ 2 of 19] Compiling Data.Metrology.Z ( Data/Metrology/Z.hs, dist/build/Data/Metrology/Z.o )

Data/Metrology/Z.hs:70:3: error:
    Illegal visible type application ‘@PSym0’
      Perhaps you intended to use TypeApplications
   |
70 | $(singletons [d| data Z = Zero | S Z | P Z deriving Eq |])
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Data/Metrology/Z.hs:70:3: error:
    Illegal visible type application ‘@SSym0’
      Perhaps you intended to use TypeApplications
   |
70 | $(singletons [d| data Z = Zero | S Z | P Z deriving Eq |])
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/Data/Metrology/Z.hs
+++ b/Data/Metrology/Z.hs
@@ -10,7 +10,8 @@
 
 {-# LANGUAGE TypeFamilies, DataKinds, TypeOperators, UndecidableInstances,
              GADTs, PolyKinds, TemplateHaskell, ScopedTypeVariables,
-             EmptyCase, CPP, TypeSynonymInstances, FlexibleInstances #-}
+             EmptyCase, CPP, TypeSynonymInstances, FlexibleInstances,
+             TypeApplications #-}
 {-# OPTIONS_GHC -fno-warn-missing-signatures #-}
 
 -----------------------------------------------------------------------------

